src/main/scala/ch/epfl/bluebrain/nexus/admin/projects/Projects.scala

```diff
@@ -234,21 +234,20 @@ class Projects[F[_]](agg: Agg[F], index: ProjectCache[F], organizations: Organiz
       .evaluateS(command.id.toString, command)
       .flatMap {
         case Right(c: Current) =>
-          toResource(c).flatMap {
-            case Right(resource) => index.replace(c.id, resource) *> F.pure(Right(resource.discard))
-            case Left(rejection) => F.pure(Left(rejection))
+          toResource(c).flatMap { resource =>
+            index.replace(c.id, resource) *> F.pure(Right(resource.discard))
           }
         case Left(rejection) => F.pure(Left(rejection))
         case Right(Initial)  => F.raiseError(UnexpectedState(command.id.toString))
       }
 
-  private def toResource(c: Current): F[ProjectResourceOrRejection] = organizations.fetch(c.organizationUuid).map {
+  private def toResource(c: Current): F[ProjectResource] = organizations.fetch(c.organizationUuid).flatMap {
```

Comment:
Everywhere else we deal with Either and here you are suddenly raising error

Reply:
it is an error if you can't transform a current state into a resource